aspnetcore/grpc/index.md

```diff
@@ -0,0 +1,185 @@
+
+Since ASP.NET Core middlewares and features share the routing pipeline, an application can be configured to serve additional request handlers such as MVC controllers in parallel with the configured gRPC services.
+
+## Integration with ASP.NET Core APIs
+
+gRPC services have full access to the ASP.NET Core features such as Dependency Injection (DI) and Logging. For example, the service implementation can resolve a logger from the DI container via the constructor:
+
+```csharp
+public class GreeterService : Greeter.GreeterBase
+{
+    public GreeterService(ILogger<GreeterService> logger)
+    {
+    }
+}
+```
+
+By default, the service implementation can resolve services with Singleton and Scoped lifetimes.
+
+### Resolve HttpContext in gRPC methods
+
+The gRPC API provides access to some underlying data of the HTTP/2 message such as the method, host, header and trailers through the `ServerCallContext` argument passed to each gRPC method:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Services/GreeterService.cs?highlight=12)]
+
+While this may be sufficient for many purposes, to ensure full access to ASP.NET Core APIs, the `HttpContext` representing the underlying HTTP/2 message can be accessed through the `GetHttpContext` extension method:
+
+```csharp
+public class GreeterService : Greeter.GreeterBase
+{
+    public override Task<HelloReply> SayHello(HelloRequest request, ServerCallContext context)
+    {
+        HttpContext httpContext = context.GetHttpContext();
+
+        return Task.FromResult(new HelloReply
+        {
+            Message = "Using https: " + httpContext.Request.IsHttps
+        });
+    }
+}
+```
+
+## gRPC service implementation lifetime
+
+By default, services will be created with a Scoped lifetime as defined in the [Dependency Injection](xref:fundamentals/dependency-injection) document. This allows the service implementation to resolve services with Scoped lifetimes, such as DBContext, from the DI container through constructor injection.
+
+> [!WARNING]
+> The Scoped lifetime of service implementation in ASP.NET Core is a behavioral difference from grpc C Core. Since a new instance of the service implementation is constructed for each request, it's no longer possible to share state between requests via instance members on the implementation type. Instead, the expectation is to store shared states in a Singleton service in the DI container and resolve it in the constructor of the gRPC service implementation.
+
+### Add a singleton service
+
+To facilitate the transition from grpc C Core implementation to ASP.NET Core, it is possible to change the service lifetime of the service implementation from Scoped to Singleton. This involves adding an instance of the service implementation to the DI container:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddGrpc();
+    services.AddSingleton(new GreeterService());
```

Comment:
Does this work? We need a functional test for this scenario

Reply:
Yea, I was going to suggest removing this all up. This section of the doc seems unnecessary...

Reply:
We should have an article on migrating from Grpc.Core to Grpc.AspNetCore. This content would be good there.

Reply:
Moved to migration doc.